src/structures/gdt.rs

```diff
@@ -61,28 +100,29 @@ impl GlobalDescriptorTable {
 }
 
 impl<const MAX: usize> GlobalDescriptorTable<MAX> {
-    /// Creates an empty GDT which can hold `MAX` number of [`Descriptor`]s.
+    /// Creates an empty GDT which can hold `MAX` number of [`Entry`]s.
     #[inline]
     pub const fn empty() -> Self {
         // TODO: Replace with compiler error when feature(generic_const_exprs) is stable.
         assert!(MAX > 0, "A GDT cannot have 0 entries");
         assert!(MAX <= (1 << 13), "A GDT can only have at most 2^13 entries");
+        const NULL: Entry = Entry::new(0);
         Self {
-            table: [0; MAX],
+            table: [NULL; MAX],
             len: 1,
         }
     }
 
-    /// Forms a GDT from a slice of `u64`.
+    /// Forms a GDT from a slice of raw [`Entry`] values.
     ///
     /// # Safety
     ///
     /// * The user must make sure that the entries are well formed
     /// * Panics if the provided slice has more than `MAX` entries
     #[inline]
-    pub const unsafe fn from_raw_slice(slice: &[u64]) -> Self {
+    pub const unsafe fn from_raw_entries(slice: &[u64]) -> Self {
```

Comment:
The docs were changed to say `Entry`, but it still takes `&[u64]`.

Reply:
Whoops, I'll change the docs back. I think we want to take `u64`s here. As if you're dealing with raw values, they will probably be `u64`s.